File: python/local.py
import os
import time

statbuf = os.stat("test_file.txt")
print("File size (in bytes): " + str(statbuf.st_size))
print(
    "Last modified time: "
    + time.strftime("%a, %d %b %Y %X GMT", time.gmtime(int(statbuf.st_mtime)))
)

fd = os.open("test_file.txt", os.O_RDWR)
buf = os.read(fd, 5)
print(str(len(buf)) + " " + buf.decode())
os.lseek(fd, 0, os.SEEK_SET)
buf2 = os.read(fd, 6)
print(str(len(buf2)) + " " + buf.decode() + buf2.decode())
os.write(fd, b"word")
os.ftruncate(fd, 30)
os.fsync(fd)
os.close(fd)

os.rename("test_file.txt", "renamed_file.txt")
os.unlink("deleted_file.txt")

namelist = os.scandir(".")
for entry in namelist:
    if entry.is_dir():
        print(entry.name + "/")
    if entry.is_file():
        print(entry.name)

File: c/local.c
#define _GNU_SOURCE // needed for ftruncate
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <sys/stat.h>
#include <time.h>
#include <dirent.h>
#include <stdlib.h>

int main(void)
{
    char date[64], buf[200] = {0}, *entry;
    const char *name;
    struct stat statbuf;
    struct tm *time_info;
    int fd, i, n;
    ssize_t status;
    struct dirent **namelist;

    stat("test_file.txt", &statbuf);
    printf("File size (in bytes): %ld\n", statbuf.st_size);
    time_info = gmtime(&statbuf.st_mtime);
    strftime(date, sizeof(date), "%a, %d %b %Y %X GMT", time_info);
    printf("Last modified time: %s\n", date);

    fd = open("test_file.txt", O_RDWR);
    status = read(fd, buf, 5);
    printf("%ld %s\n", status, buf);
    lseek(fd, 0, SEEK_SET);
    status = read(fd, buf + 5, 6);
    printf("%ld %s\n", status, buf);
    write(fd, "word", 4);
    ftruncate(fd, 30);
    fsync(fd);
    close(fd);

    rename("test_file.txt", "renamed_file.txt");
    unlink("deleted_file.txt");

    n = scandir(".", &namelist, NULL, alphasort);
    for (i = 0; i < n; i++) {
        name = namelist[i]->d_name;
        stat(name, &statbuf);
        if (S_ISDIR(statbuf.st_mode)) {
            printf("%s/\n", name);
        }
        if (S_ISREG(statbuf.st_mode)) {
            printf("%s\n", name);
        }
    }
    free(namelist);

    return 0;
}
